compiler/llvm/clangUtil.cpp

```diff
@@ -3435,38 +3496,92 @@ void makeBinaryLLVM(void) {
   // Setup and run LLVM passes to emit a .o file to outputOfile
   {
-    llvm::legacy::PassManager emitPM;
 
-    emitPM.add(createTargetTransformInfoWrapperPass(
-               info->targetMachine->getTargetIRAnalysis()));
+    bool disableVerify = !developer;
+
+    if (gCodegenGPU == false) {
+      llvm::raw_fd_ostream outputOfile(moduleFilename, error, flags);
+      if (error || outputOfile.has_error())
+        USR_FATAL("Could not open output file %s", moduleFilename.c_str());
 
 #if HAVE_LLVM_VER >= 100
-    llvm::CodeGenFileType FileType = llvm::CGFT_ObjectFile;
+      llvm::CodeGenFileType FileType = llvm::CGFT_ObjectFile;
 #else
-    llvm::TargetMachine::CodeGenFileType FileType =
-      llvm::TargetMachine::CGFT_ObjectFile;
+      llvm::TargetMachine::CodeGenFileType FileType =
+        llvm::TargetMachine::CGFT_ObjectFile;
 #endif
 
-    bool disableVerify = ! developer;
+      {
+        llvm::legacy::PassManager emitPM;
+
+        emitPM.add(createTargetTransformInfoWrapperPass(
+                   info->targetMachine->getTargetIRAnalysis()));
+
 #if HAVE_LLVM_VER > 60
-    info->targetMachine->addPassesToEmitFile(emitPM, outputOfile,
-                                             nullptr,
-                                             FileType,
-                                             disableVerify);
+        info->targetMachine->addPassesToEmitFile(emitPM, outputOfile,
+                                                 nullptr,
+                                                 FileType,
+                                                 disableVerify);
 #else
-    info->targetMachine->addPassesToEmitFile(emitPM, outputOfile,
-                                             FileType,
-                                             disableVerify);
+        info->targetMachine->addPassesToEmitFile(emitPM, outputOfile,
+                                                 FileType,
+                                                 disableVerify);
 #endif
 
-    // Run the passes to emit the .o file now!
-    emitPM.run(*info->module);
-    outputOfile.close();
+        emitPM.run(*info->module);
+
+      }
+      outputOfile.close();
+
+    } else {
+
+      llvm::CodeGenFileType asmFileType =
+        llvm::CodeGenFileType::CGFT_AssemblyFile;
+
+      llvm::raw_fd_ostream outputASMfile(asmFilename, error, flags);
+
+      {
+
+        llvm::legacy::PassManager emitPM;
+
+        emitPM.add(createTargetTransformInfoWrapperPass(
+                   info->targetMachine->getTargetIRAnalysis()));
+
+        info->targetMachine->addPassesToEmitFile(emitPM, outputASMfile,
+                                                 nullptr,
+                                                 asmFileType,
+                                                 disableVerify);
+
+        emitPM.run(*info->module);
+
+      }
+
+      outputASMfile.close();
+
+      std::string ptxCmd = std::string("/usr/local/cuda/bin/ptxas -m64 --gpu-name ") +
```

Comment:
These paths shouldn't be hardcoded like this. We could require these tools to be in PATH (so here we just run it as `ptxas`) and/or we could have a couple of places we try to find the cuda installation to detect it.